### ISSUE TITLE
napari cli: enable --with option for npe2 widgets 

### ISSUE THREAD

**Issue by nclack** [napari/napari#3805]
## Description

Right now trying to launch an  npe2 dock widget from the cli fails. For example:

```sh
napari --with napari-animation
```

(with napari-animation installed as an npe2 plugin)

gives

```
Traceback (most recent call last):
  File "/Users/nclack/opt/miniforge3/envs/dev-napari-npe2-console/bin/napari", line 33, in <module>
    sys.exit(load_entry_point('napari', 'console_scripts', 'napari')())
  File "/Users/nclack/src/CZI/napari/napari/__main__.py", line 449, in main
    _run()
  File "/Users/nclack/src/CZI/napari/napari/__main__.py", line 327, in _run
    viewer.window.add_plugin_dock_widget(pname)
  File "/Users/nclack/src/CZI/napari/napari/_qt/qt_main_window.py", line 655, in add_plugin_dock_widget
    widget_name = list(plugin_manager._dock_widgets[plugin_name])[0]
KeyError: 'napari-animation'
```

## References

I ran into this while looking into this comment from @jni.  This issue focuses on the idea of pre-loading a plugin widget using the `--with` option, but doesn't address the `napari path/to/file.npy`.  See discussion below for more.

> Can we update this to be a single command that "executes" the plugin? e.g. `napari path/to/file.npy`? That's not a super 
> interesting example, but I always have to look up the equivalent command for pre-loading a plugin widget.

_Originally posted by @jni in https://github.com/napari/napari/pull/3769#discussion_r767255994_

**Comment by tlambert03** [comment]
will comment over on the PR as well, but I want to point out that there's 2 behaviors currently supported with npe1.  and I _think_ juan was referring to the second one (which is not fixed in #3807)... but I could be wrong:

1. launching napari with a widget already opened using `napari --with some-plugin`  (implemented in #3807)
2. "executing" a plugin module with `napari path/to/file.npy`, as implemented in https://github.com/napari/napari/pull/2503. Which is very useful for testing a plugin without installation.  When @jni said "a single command that "executes" the plugin? e.g. `napari path/to/file.npy`" ...   I interpreted it to mean that.

@jni can weigh in on whether all he wanted was the `--with` flag to work with `npe2`, or if he meant the latter

**Comment by nclack** [comment]
thanks.  I'll edit the issue to clarify.  This issue covers only the `--with` behavior.
